packages/graphback-runtime-mongodb/src/queryBuilder.ts

```diff
@@ -0,0 +1,137 @@
+import {AdvancedFilter} from '@graphback/runtime';
+
+const AND_FIELD = 'and';
+const OR_FIELD = 'or';
+const NOT_FIELD = 'not';
+
+// Operators that work with simple substitution :P
+const operatorMap = {
+    [AND_FIELD]: '$and',
+    [OR_FIELD]: '$or',
+    [NOT_FIELD]: '$nor',
+    ne: '$ne',
+    eq: '$eq',
+    le: '$lte',
+    lt: '$lt',
+    ge: '$gte',
+    gt: '$gt',
+}
+
+type OperatorTransform = [string, any][];
+
+type OperatorTransformFunction = (value: any) => OperatorTransform;
+
+// A map of functions to transform mongodb incompatible operators
+// Each function returns pairs of a key and an object for that key
+const operatorTransform: {
+    [operator: string]: OperatorTransformFunction
+} = {
+    between: (values: [any, any]): OperatorTransform => {
+        values.sort();
+
+        return [
+            [operatorMap.ge, values[0]],
+            [operatorMap.le, values[1]]
+        ]
+    },
+    nbetween: (values: [any, any]): OperatorTransform => {
+
+        values.sort();
+
+        return [
+            [
+                "$not", 
+                {
+                    [operatorMap.ge]: values[0],
+                    [operatorMap.le]: values[1],
+                }
+            ],
+            [
+                "$exists",
+                true
+            ]
+        ]
+    },
+    contains: (value: string): OperatorTransform => {
+        return [
+            [operatorMap.eq, new RegExp(value, 'g')]
+        ]
+    },
+    startswith: (value: string): OperatorTransform => {
+        return [
+            [operatorMap.eq, new RegExp(`^${value}`, 'g')]
+        ]
+    },
+    endswith: (value: string): OperatorTransform => {
+        return [
+            [operatorMap.eq, new RegExp(`${value}$`, 'g')]
+        ]
+    },
+}
+
+// Function to check if variable is primitive or a map
+function isPrimitive(test: any): boolean {
+    return ((test instanceof RegExp) || (test !== Object(test)));
+};
+
+
+function traverse(filter: any): any {
+
+    Object.keys(filter).forEach((key: string) => {
+        if (filter.hasOwnProperty(key)) {
+            // Transform the operators to mongodb operators
+
+            // Check if it can be directly substituted
+            if (operatorMap[key]) {
+
+                // Substitute it with the mongodb operator
+                filter[operatorMap[key]] = filter[key];
+                /*eslint-disable-next-line*/
+                delete filter[key];
```

Comment:
https://palantir.github.io/tslint/rules/no-dynamic-delete/

> Deleting dynamically computed keys is dangerous and not well optimized.

It looks like this forEach is modifying the original `filter`, so maybe a map could be used instead?